concent_api/core/views.py

```diff
@@ -370,8 +393,23 @@ def validate_golem_message_reject(data):
         if data.compute_task_def['task_id'] == '':
             raise Http400("task_id cannot be blank.")
 
-        if not isinstance(data.compute_task_def['deadline'], int):
-            raise Http400("Wrong type of deadline field.")
+        data.compute_task_def['deadline'] = validate_int_value(data.compute_task_def['deadline'])
+
+
+def validate_int_value(value):
+    """
+    Checks if value is integer, if not it tries to cast it as integer.
+    Then it checks if value is non-negative integer.
+
+    """
+    if not isinstance(value, int):
+        try:
+            value = int(value)
+        except (ValueError, TypeError):
+            raise Http400("Wrong type of deadline field, expected integer or value that can be casted to integer.")
```

Comment:
The field is not necessarily called `deadline`. This might be another integer field